node/src/network.rs

```diff
@@ -213,6 +222,45 @@ impl<const N: usize> crate::Network for Kadcast<N> {
         Ok(())
     }
 
+    /// Broadcast a GetResource request.
+    ///
+    /// By utilizing the randomly selected peers per bucket in Kadcast, this
+    /// broadcast does follow the so-called "Flood with Random Walk" blind
+    /// search (resource discovery).
+    ///
+    /// A receiver of this message is supposed to look up the resource and
+    /// either return it or, if not found, rebroadcast the message to the next
+    /// Kadcast bucket
+    ///
+    /// * `ttl_as_sec` - Defines the lifespan of the request in seconds
+    ///
+    /// * `hops_limit` - Defines maximum number of hops to receive the request
+    async fn flood_request(
+        &self,
+        msg_inv: &Inv,
+        ttl_as_sec: Option<u64>,
+        hops_limit: u16,
+    ) -> anyhow::Result<()> {
+        let ttl_as_sec = ttl_as_sec.map_or_else(
+            || u64::MAX,
+            |v| {
+                SystemTime::now()
+                    .duration_since(UNIX_EPOCH)
+                    .unwrap_or_default()
+                    .as_secs()
+                    + v
+            },
+        );
+
+        self.broadcast(&Message::new_get_resource(GetResource::new(
```

Comment:
Currently, the broadcast (kadcast) is highly inefficient for a flood request due to  the `redundancy factor β per bucket` which by default is 3.

In other words, on  broadcasting a GetResource (hops_limit=1), it will reach at least 3*N nodes where is N is the number of buckets that have a peer.

In a network of 10K nodes, this behavior might lead to hundreds of duplicated responses to the `GetResource` msg.

Reply:
Currently, the solution will be:

On sending a flooding request, the peer will randomly retrieve N alive peers which will receive the request. Any of the receivers will search for the resource and  will send the result back to the requester. If the resource is not found, the receiver will randomly retrieve M alive peers and will re-send the request to them.

```
N - 8 peers
M - 1 peer
hops_limit = 64
```